### PR TITLE
Changes required for python 3

### PR DESCRIPTION
**Pull request by timj**
Uses future
- Use utf-8 encoding for file handles
- Use ascii encoding for SHA1
- Output from subprocess must be decoded
- Iterations explicitly stop with b'' not u''
- itervalues() et al -> values()
- Inherit from object and Exception consistently.
- Sorted uses a key function not a cmp function
- Explicitly use list(keys) in one loop.

### CODE REVIEW

**Comment by timj** [comment]
Travis can't build this at the moment because it now depends on the `future` python package.

**Comment by timj** [comment]
Everything seems to work now on Python 2 and Python 3 with Travis testing both.

File: python/lsst/ci/build.py [hunk 1]
```diff
@@ -194,24 +194,24 @@ def _build_product(self, product, progress):
         os.chmod(buildscript, st.st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
 
         # Run the build script
-        with open(logfile, 'w') as logfp:
+        with open(logfile, 'w', encoding="utf-8") as logfp:
```
Comment (jhoblitt) [inline python/lsst/ci/build.py]: Shouldn't this be single quotes?

Comment (timj) [inline python/lsst/ci/build.py]: you mean for consistency? Probably I hadn't even noticed that the rest of the code uses single quotes as I always use double quotes (from my perl heritage).

Comment (jhoblitt) [inline python/lsst/ci/build.py]: I believe pep8 says something like "pick a rule and stick with it" for quoting style.

File: python/lsst/ci/prepare.py [hunk 2]
```diff
@@ -66,19 +70,19 @@ def __init__(self, productsList, buildID=None):
 
     def toFile(self, fileObject):
```
Comment (jhoblitt) [inline python/lsst/ci/prepare.py]: `fileObject` is still in camelCase.

**Comment by jhoblitt** [comment]
This looks good other than the trivial style inconsistencies I noted.  Don't forget to fix the lsstsw branch name before merging...

File: python/lsst/ci/prepare.py (hunk 2 above)
Comment (timj) [inline python/lsst/ci/prepare.py]: I think DM style requires that (in the modified PEP-8 rules).

Oh, the argument, will fix. `flake8` only looks for assignments.

**Comment by timj** [comment]
I think I'm ready to merge this but lsst/lsstsw#111 has to be merged first (and I'm just waiting on the linux pinned package list).

**Pull request merged**